server/controllers/admin/fundingSources.js

```diff
@@ -0,0 +1,73 @@
+const db = require('../../lib/db');
+
+module.exports = {
+  create,
+  read,
+  update,
+  detail,
+  delete : remove,
+};
+
+// add a new tag
+function create(req, res, next) {
+  const sql = `INSERT INTO funding_source SET ?`;
+  const data = req.body;
+  data.uuid = data.uuid ? db.bid(data.uuid) : db.uuid();
+  db.exec(sql, data)
+    .then(() => {
+      res.sendStatus(201);
+    }).catch(next);
+}
+
+// update tag information
+function update(req, res, next) {
+  const sql = `UPDATE funding_source SET ?  WHERE uuid =?`;
+  const data = req.body;
+  delete data.uuid;
+  const uuid = db.bid(req.params.uuid);
+
+  db.exec(sql, [data, uuid])
+    .then(() => {
+      res.sendStatus(200);
```

Comment:
As above, the `update` route should return `200 (OK), 404 (NOT FOUND), 400 (BAD REQUEST)`, depending on the result.

The update function should return a JSON object with the full resource.